lib/aerospike/cluster/find_node.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Aerospike
+  class Cluster
+    # Find node in cluster by name
+    module FindNode
+      class << self
+        def call(cluster, peers, node_name)
+          node = cluster.find_node_by_name(node_name) || peers.nodes[node_name]
+          return if node.nil?
+          node.tap do |n|
+            n.increase_reference_count!
```

Comment:
With my limited knowledge, that makes sense since the node isn't technically added to the cluster if found in peers. I'm not super happy about this (somewhat unexpected) side effect of this method. I'll try and re-arrange this a bit. Let me know if you have any new thoughts

Reply:
It seems like the Java client also increases the reference count when the node is found in peers, so I'm keeping it until you have further insights

https://github.com/aerospike/aerospike-client-java/blob/7352c15c04015f4bd4f6985e138fa931179fb5d7/client/src/com/aerospike/client/cluster/Node.java#L379-L396